src/wallOverlap.cpp

```diff
@@ -182,6 +171,7 @@ int64_t WallOverlapComputation::getApproxOverlapArea(const Point from, const Poi
         const int64_t other_to_proj = dot(other_to - to, vec) / vSize(vec);
 
         const int64_t overlap_length_2 = to_proj + other_to_proj; //Twice the length of the overlap area, alongside the lines.
+        const int64_t overlap_width_2 = line_width * 2 - std::sqrt(LinearAlg2D::getDist2FromLineSegment(from, other_to, to)) - std::sqrt(LinearAlg2D::getDist2FromLineSegment(other_from, to, other_to));
```

Comment:
This distance should already be stored in the link, which is read by `handlePotentialOverlap` and passed on via the `from_dist` and `to_dist` parameters here. If that distance is wrong, we should fix it when creating the link, not re-compute it here.

Reply:
But it's not the distances between to/other_to and from/other_from that I am computing here.

Reply:
Indeed, you're computing the difference between the distances from `to` to the other line and from `other_to` to this line. That seems kind of random, now that you mention it. What's that for, and can you document it a bit better then? Why can't we store this one in the link?

Reply:
Yes, I am computing the distances from each of the line ends to the other line because I want to determine the overlap area for exactly the region of overlap which I am calculating as the overlap length times the average distance between the two lines (in the overlap region).
That's a good one about the documentation, this whole subsystem (overlap compensation) is so badly documented nobody but the original author(s) have a chance of of understanding how any of it works (and even they probably can't remember what they did before).
As to why this distance can't be stored in the link, I honestly don't know because I don't understand how the link code works.

Reply:
I checked that math and it seems to be correct (I arrived at the same result calculating the area of two right triangles). It'd be good to document this area calculation still, even if the rest isn't documented well. And I don't know if we can accept it if it's not pre-calculated in the link.

Reply:
OK, I will document this calculation. Well, this whole PR is really just for discussing some of the flaws in the overlap compensation. I'm using this PR and I believe it produces better results (at least visually on the layer view) than the original code. Whether you decide to incorporate it (or even just parts of it) really doesn't concern me. At least you are now aware of some of the problems. Every now and again when I am feeling up to the challenge I try to understand this part of Cura a little better but I still don't really "get it"!